selenium_tests/publish_courses/publish_courses_tests.py

```diff
@@ -0,0 +1,92 @@
+from bulk_utilities.bulk_course_settings import BulkCourseSettingsOperation
+
+from selenium_common.canvas.canvas_course_page_object import CanvasCoursePage
+
+from selenium_tests.publish_courses.publish_courses_base_test_case import \
+    PublishCoursesBaseTestCase
+from selenium_tests.publish_courses.page_objects.main_page import \
+    MainPageObject
+
+
+class PublishCoursesTests(PublishCoursesBaseTestCase):
+
+    def test_publish_courses(self):
+        """TLT-1805: This test case goes through the workflow of publishing
+        all courses in a specific term using the Publish Courses tool """
+
+        main_page = MainPageObject(self.driver)
+
+        #  Select a term from the term dropdown
+        term_with_unpublished_courses = self.term['test_terms'][
+            'term_with_unpublished_courses']
+        main_page.select_term(term_with_unpublished_courses)
+
+        # If the Publish Courses button is enabled, click on it.
+        if main_page.is_publish_all_button_enabled():
+            main_page.publish_courses()
+        else:
+            #  If "Publish all" button is not enabled, unpublish courses
+            #  first to ensure that the publish functionality can be tested.
+            self.bulk_unpublish_canvas_sites()
+
+            #  Refresh the page to see the changes
+            self.refresh_page(term_with_unpublished_courses)
```

Comment:
See note about re-selecting term instead of refresh_page.

Also note that it may take a few seconds for all terms to be processed, but probably by the time you requery (even with the quicker re-select term method) at least one term will be unpublished, which will enable the button).